dsl/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/dsl/jpql/Jpql.kt

```diff
@@ -602,6 +602,22 @@ open class Jpql : JpqlDsl {
         return Expressions.floor(value.toExpression())
     }
 
+    /**
+     * Creates an expression that represents the index of the value in an ordered list.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any, V, S : Collection<V>> index(expr: KProperty1<T, @Exact S>): Expression<Int> {
+        return Expressions.index(Paths.path(expr))
+    }
+
+    /**
+     * Creates an expression that represents the index of the value in an ordered list.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T, S : Collection<T>> index(path: Pathable<S>): Expression<Int> {
```

Comment:
It seems that the INDEX function can take an alias of join as a parameter, so shouldn't the type of the parameter be Expressionable?

Reply:
Ahh, you are right 😄